app/src/components/SoldArtworkList.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import SoldArtworkItem from './SoldArtworkItem';
+import CardColumns from 'react-bootstrap/CardColumns';
+import { ContractProps } from '../helper/eth';
+
+interface SoldArtworkListState {
+  balance: number;
+  tokenIds: Array<number>;
+  prices: Array<number>;
+  newOwners: Array<string>;
+}
+
+class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListState> {
+  constructor (props: ContractProps) {
+    super(props);
+    this.state = {
+      balance: 0,
+      tokenIds: [],
+      prices: [],
+      newOwners: [],
+    };
+  }
+
+  componentDidMount (): void {
+    this.loadSales();
+  }
+
+  shouldComponentUpdate (): boolean {
+    this.loadSales();
+    return true;
+  }
+
+  async loadSales (): Promise<void> {
+    const governance = this.props.contracts.Governance;
+    const currentAccount = this.props.accounts[0];
+    let eventsFound = 0;
+    const tokensArray: any[] = [];
+    const priceArray: any[] = [];
+    const newOwnerArray: any[] = [];
+
+    await governance.getPastEvents(
+      'RecordARR',
+      (errors: any, events: any) => {
+        if (!errors) {
+          for (let i = 0; i < events.length; i++) {
+            if (events[i].returnValues.from === currentAccount) {
```

Comment:
I think this only filters for sales where the user has owned the artifact. Do we have a separate task for displaying sales that a dealer/auction house has done?

Reply:
We do not currently have a separate task but you are right that we should do this and don’t yet: I think it should be a separate task as it requires refactoring of our emitted events so that the approval address is something emitted on every single transfer

Reply:
Though I’m happy to pick that task up as soon as this one is complete, I feel that would make the most sense